src/Microsoft.AspNetCore.Http.Abstractions/Extensions/EndpointModel.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// A base class for modeling <see cref="Endpoint"/>.
+    /// </summary>
+    public abstract class EndpointModel
```

Comment:
Why doesn’t the builder have the Build method? Why are these different types?

Reply:
EndpointModel is used to create one endpoint, e.g. an action on a controller.

IEndpointConventionBuilder is used to register conventions that are applied to multiple endpoints as they are being created. The models are passed to `Action<EndpointModel>` that applies the convention changes, e.g. all endpoints created by MVC with a conventional route pattern should have IAuthorizeData metadata.

Reply:
Thought about how to explain this more:
* EndpointModel: mutable version of Endpoint that are created when a datasource initializes. Needed because we want to apply changes to endpoints as they are built with a convention
* IEndpointConventionBuilder: start up time way of registering conventions. These conventions are then run when a datasource initializes. Data source creates models, conventions run on them, then models are turned into immutable endpoints